### ISSUE TITLE
Can not use the audioplayer just before calling an audiorecorder method (it works on Simulator though)

### ISSUE THREAD

**Issue by mouneyrac** [jsierles/react-native-audio#74]
It can be reproduce on the current AudioExample on an iPhone 5c (everything does work fine on the simulator - I suspect the simulator been too fast to have the problem):

Add the AudioPlayer:
import {AudioRecorder, AudioPlayer, AudioUtils} from 'react-native-audio';

Then in 
_record() {
    AudioPlayer.play(AudioUtils.MainBundlePath + '/start.m4a'); // New line - the file is added to the root folder of the xcode project checking copy and target.
    AudioRecorder.startRecording();
    this.setState({recording: true, playing: false});
  }

On simulator, no issues. On iPhone5c the sound is played but then the timer stay at 0. 

Additionally, sometime the iPhone5c records properly (with the timer working one time), but once the recording finishes I played another sound (success.m4a) and then when I try to press Play, I have the same problem. The recording is not played and the onprogress is called only once and the currentTime is equal 0 when you console.log in the onProgress - which is weird as when things work the currentTime is never equal to 0 when you console.log it in the onProgress. (all is still working fine in the simulator)

I confirm that without playing a sound before calling a AudioRecording function, the AudioExample works perfectly on iPhone5c.

I am thinking making the play function of the AudioPlayer returning a promise but it looks more like a patch than fixing the original issue (maybe it will not even work though, I am looking on how to create a promise in JS :)). I thought that maybe you would know what happen and help me to go in the right direction to fix it :)

**Comment by jsierles** [comment]
I'll take a look, thanks for the report!

**Comment by mouneyrac** [comment]
Thanks Joshua, actually after thinking about the promise, it may be the correct solution anyway as in fact without promise the recording will start when the sound is still playing. Moreover there may be other issues. If you were fixing all problem of my use case (just for the info I am trying to reproduce the camera iOS app win only audio with same sounds when you press the buttons) I think you would end up with implementing a lot of react-native-sound. I am just testing react-native-sound and it supports promise and pre-loading of the sound which I think is important as it cuts a part of the sound.

At the end I think we can fix my issue changing the sessionCategory for AVAudioSessionCategoryPlayAndRecord. For anyone interested (obviously Joshua knows), in react-native-audio, you can change the the sessionCategory when calling AudioPlayer.play(AudioUtils.MainBundlePath + '/start.m4a', {sessionCategory: 'AVAudioSessionCategoryPlayAndRecord'); You will find the constant in https://developer.apple.com/library/ios/documentation/AVFoundation/Reference/AVAudioSession_ClassReference/#//apple_ref/doc/constant_group/Audio_Session_Categories

I guess we can close this as changing the sessionCategory "fix" the report (or put a label 'deferred' or something like that). I suppose the best solution would be merging both react-native-sound and the recording part of react-native-audio together.

Thank you for you fast communication and help Joshua!

**Comment by jsierles** [comment]
I think it may be a good time to consider recommending react-native-sound for playback, or maybe integrating directly with it.

I don't use this module currently, but if I did, I would not be using its playback features. So my vote would be to deprecate playback, recommend react-native-sound, then look into possible integration.

**Comment by jsierles** [comment]
Added an issue https://github.com/zmxv/react-native-sound/issues/49

**Comment by mouneyrac** [comment]
Thanks Joshua it would be great to have all the community effort under one audio project (if someone is willing to give his time for maintaining it / is paid by his company to do it). Thank you also to let us know that you don't use react-native-audio - it is always good to know the situation of the main plugin maintainer. I am myself an author of a relatively "popular" plugin (but not for react native)  that I haven't been using it much since I released it, so I understand the implication :)

Cheers.

**Comment by jsierles** [comment]
Closing this out in favor of a future PR to merge with react-native-sound.
